Breakage with Bazel@HEAD

`rules_haskell` is broken on the continuous integration tests of Bazel @ HEAD:

https://buildkite.com/bazel/bazel-at-head-plus-downstream/builds/1903#81e5c204-6ddf-4452-85a5-65b950176c5a

This breakage is probably caused by https://buildkite.com/bazel/bazel-at-head-plus-downstream/builds/1903#81e5c204-6ddf-4452-85a5-65b950176c5a that calls test binaries in external repositories not using their legacy `external/$REPO/$REPO_RELATIVE_PATH` path but the canonical `../$REPO/$REPO_RELATIVE_PATH` one.

I haven't looked into what the mechanism of the breakage is; my guess is that test binaries try to find data files relative to $0 and that logic needs to be updated

This issue is also available in our continuous-integration repository as bazelbuild/continuous-integration#1090 .

After https://github.com/bazelbuild/bazel/commit/7149f578006a4ad0d51df69830a6986749b34df5 the error is different, see below.

Full log at https://buildkite.com/bazel/bazel-at-head-plus-downstream/builds/1924#8f955628-60b7-4249-8925-a7daabe7efd1

```
exe/Main.hs:671:24: error:
--
  | • Exception when trying to run compile-time code:
  | /var/lib/buildkite-agent/.cache/bazel/_bazel_buildkite-agent/18c25b4026b5f422e1efd6536818b1cd/execroot/rules_haskell/external/rules_haskell_ghc_linux_amd64/bin/../lib/settings: openFile: does not exist (No such file or directory)
  | Code: makeGhcVersionChecker (pure <$> getLibdir)
  | • In the Template Haskell splice
  | $$(makeGhcVersionChecker (pure <$> getLibdir))
  | In the expression: $$(makeGhcVersionChecker (pure <$> getLibdir))
  | In an equation for ‘ghcVersionChecker’:
  | ghcVersionChecker = $$(makeGhcVersionChecker (pure <$> getLibdir))
  | \|
  | 671 \| ghcVersionChecker = $$(makeGhcVersionChecker (pure <$> getLibdir))
  | \|                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  | Traceback (most recent call last):
  | File "/var/lib/buildkite-agent/.cache/bazel/_bazel_buildkite-agent/6d66f770f087e3aa3ec33d66ffb996f0/sandbox/linux-sandbox/1895/execroot/rules_haskell/bazel-out/host/bin/haskell/cabal_wrapper.runfiles/rules_haskell/haskell/cabal_wrapper.py", line 194, in <module>
  | run([runghc] + runghc_args + [setup, "build", "--verbose=0", "--builddir=" + distdir])
  | File "/var/lib/buildkite-agent/.cache/bazel/_bazel_buildkite-agent/6d66f770f087e3aa3ec33d66ffb996f0/sandbox/linux-sandbox/1895/execroot/rules_haskell/bazel-out/host/bin/haskell/cabal_wrapper.runfiles/rules_haskell/haskell/cabal_wrapper.py", line 42, in run
  | subprocess.run(cmd, check=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE, *args, **kwargs)
  | File "/usr/lib/python3.6/subprocess.py", line 438, in run
  | output=stdout, stderr=stderr)



```
exe/Main.hs:671:24: error:
--
  | • Exception when trying to run compile-time code:
  | /var/lib/buildkite-agent/.cache/bazel/_bazel_buildkite-agent/18c25b4026b5f422e1efd6536818b1cd/execroot/rules_haskell/external/rules_haskell_ghc_linux_amd64/bin/../lib/settings: openFile: does not exist (No such file or directory)
  | Code: makeGhcVersionChecker (pure <$> getLibdir)
  | • In the Template Haskell splice
  | $$(makeGhcVersionChecker (pure <$> getLibdir))
  | In the expression: $$(makeGhcVersionChecker (pure <$> getLibdir))
  | In an equation for ‘ghcVersionChecker’:
  | ghcVersionChecker = $$(makeGhcVersionChecker (pure <$> getLibdir))
  | \|
  | 671 \| ghcVersionChecker = $$(makeGhcVersionChecker (pure <$> getLibdir))
  | \|                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  | Traceback (most recent call last):
  | File "/var/lib/buildkite-agent/.cache/bazel/_bazel_buildkite-agent/6d66f770f087e3aa3ec33d66ffb996f0/sandbox/linux-sandbox/1895/execroot/rules_haskell/bazel-out/host/bin/haskell/cabal_wrapper.runfiles/rules_haskell/haskell/cabal_wrapper.py", line 194, in <module>
  | run([runghc] + runghc_args + [setup, "build", "--verbose=0", "--builddir=" + distdir])
  | File "/var/lib/buildkite-agent/.cache/bazel/_bazel_buildkite-agent/6d66f770f087e3aa3ec33d66ffb996f0/sandbox/linux-sandbox/1895/execroot/rules_haskell/bazel-out/host/bin/haskell/cabal_wrapper.runfiles/rules_haskell/haskell/cabal_wrapper.py", line 42, in run
  | subprocess.run(cmd, check=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE, *args, **kwargs)
  | File "/usr/lib/python3.6/subprocess.py", line 438, in run
  | output=stdout, stderr=stderr)
```

--- a/.bazelci/presubmit.yml
+++ b/.bazelci/presubmit.yml
@@ -5,10 +5,10 @@ tasks:
       # haskell base uses the environment locale to decode sockets
       LANG: "C.UTF-8"
     build_flags:
-      - "--build_tag_filters=-requires_zlib,-requires_lz4,-requires_doctest,-requires_c2hs,-requires_threaded_rts,-requires_shellcheck,-dont_test_with_bindist,-dont_test_on_bazelci"
+      - "--build_tag_filters=-requires_nix,-requires_lz4,-requires_shellcheck,-requires_threaded_rts,-dont_test_with_bindist,-dont_test_on_bazelci"
     build_targets:
       - "//tests/..."
     test_flags:
-      - "--test_tag_filters=-requires_zlib,-requires_lz4,-requires_doctest,-requires_c2hs,-requires_threaded_rts,-requires_shellcheck,-dont_test_with_bindist,-dont_test_on_bazelci"
+      - "--test_tag_filters=-requires_nix,-requires_lz4,-requires_shellcheck,-requires_threaded_rts,-dont_test_with_bindist,-dont_test_on_bazelci"
     test_targets:
       - "//tests/..."

